inference-engine/ie_bridges/python/src/openvino/inference_engine/ie_api.pyx

```diff
@@ -414,14 +414,14 @@ cdef class IECore:
                 net.impl = self.impl.readNetwork(model_, weights_)
         return net
 
-    cpdef ExecutableNetwork load_network(self, network: [IENetwork, str], str device_name, config=None, int num_requests=1):
+    cpdef ExecutableNetwork load_network(self, network: [IENetwork, str], device_name: [str, NoneType], config=None, int num_requests=1):
         """Loads a network that was read from the Intermediate Representation (IR) to the plugin with specified device name
         and creates an :class:`ExecutableNetwork` object of the :class:`IENetwork` class.
         You can create as many networks as you need and use them simultaneously (up to the limitation of the hardware
         resources).
         
         :param network: A valid :class:`IENetwork` instance. Model file name .xml, .onnx can also be passed as argument
-        :param device_name: A device name of a target plugin
+        :param device_name: A device name of a target plugin, if none it will use AUTO device.
```

Comment:
Do we actually need such an update for legacy cython API? When a device can be omitted it looks ok, but here I don't understand why `None` is more elegant than `"AUTO"`

Reply:
Set “AUTO” and None will enter difference C++ APIs, in actual there are the same action.
You mean that it will be better to set device_name=None by default? So python user can omit device_name parameter like this:

exec_net = ie.load_network(net)

Reply:
@nkogteva The reason set to None here is because we want to distinguish if the user call our API with device_name="AUTO" or just without device_name parameter. For these 2 different cases, there will be different prompt messages in case of the errors. We have added it for C++ API LoadNetwork, so we want it be the same behavior for Python API even it is becoming legacy.